episodes/06b-branches.md

```diff
@@ -0,0 +1,329 @@
+---
+title: Branches
+teaching: 20
+exercises: 0
+---
+
+::::::::::::::::::::::::::::::::::::::: objectives
+
+- Understand why branches are useful for:
+  - Working on separate tasks in the same repository concurrently
+  - Trying multiple solutions to a problem
+  - Check-pointing versions of code
+- Merge branches back into the main branch
+
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::: questions
+
+- What are branches?
+- How can I work in parallel using branches?
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+So far we've always been working in a straight timeline.
+However, there are times when we might want to keep
+our main work safe from experimental changes we are working on.
+To do this we can use branches to work on separate tasks in parallel
+without changing our current branch, `main`.
+
+We didn't see it before but the first branch made is called `main`.
+This is the default branch created when initializing a repository and
+is often considered to be the "clean" or "working" version of a
+repository's code.
+
+We can see what branches exist in a repository by typing
+
+```bash
+$ git branch
+```
+
+```output
+* main
+```
+
+
+The `*` indicates which branch we are currently on.
+
+In this lesson, Alfredo is trying to run an analysis
+and doesn't know if it will be faster in bash or python.
+To keep his main branch safe he will use separate branches
+for both bash and python analysis.
```

Comment:
I think it would be good to keep the "flavour" of Alfredo as a chef keeping his recipes under version control. So perhaps we could say he's experimenting with a new version of his famous strawberry cream cake in which he uses mango and less sugar, or something of the kind?

Reply:
That's pretty much what I would suggest. But I think we need a good discussion on whether to do one branch or two branches, I'm not certain of my opinion.